python/cudf/cudf/_lib/column.pyx

```diff
@@ -431,28 +459,38 @@ cdef class Column:
         # After call to release(), c_col is unusable
         cdef column_contents contents = move(c_col.get()[0].release())
 
-        data = DeviceBuffer.c_from_unique_ptr(move(contents.data))
-        data = as_buffer(data)
+        data = as_buffer(
+            DeviceBuffer.c_from_unique_ptr(move(contents.data)),
+            exposed=data_ptr_exposed
+        )
 
         if null_count > 0:
-            mask = DeviceBuffer.c_from_unique_ptr(move(contents.null_mask))
-            mask = as_buffer(mask)
+            mask = as_buffer(
+                DeviceBuffer.c_from_unique_ptr(move(contents.null_mask)),
+                exposed=data_ptr_exposed
+            )
         else:
             mask = None
 
         cdef vector[unique_ptr[column]] c_children = move(contents.children)
-        children = ()
+        children = []
         if c_children.size() != 0:
-            children = tuple(Column.from_unique_ptr(move(c_children[i]))
-                             for i in range(c_children.size()))
+            # Because of a bug in Cython, we cannot set the optional
```

Comment:
I haven't seen it reported. 
I will try to make a reproducer but basically, in the following code `Column.from_unique_ptr` is called as if `data_ptr_exposed=False`.
```python
tuple(
    Column.from_unique_ptr(move(c_children[i]), data_ptr_exposed=True) for i in range(c_children.size())
)
```

Anyone encountered something similar?

Reply:
Hmm I wonder if this is due to the default parameter value for `from_unique_ptr`. For testing, what happens if you make a new function `from_unique_ptr2` that is identical but doesn't have the default and use that here instead (I'm suggesting a copy so that you don't have to modify every other call site just for testing).

Reply:
Then it works.

Reply:
OK so there's some bug with the default parameter. I'll see if I can come up with some MWE that reproduces the error, but for now this is fine.

Reply:
Thanks @vyasr